### ISSUE TITLE
[Feature Request] Parameters in memorized roll?

### ISSUE THREAD

**Issue by aturtledoesbite** [ArtemGr/Sidekick#127]
Would it be possible to make a memorized roll that takes a number for part of its roll?

Example input: 
`/r $sixSidedDice X = Xd6`
`/r $sixSidedDice 5`

Example output for above:
`5d6  = (3+1+6+2+4) = 16`

I figure this would be useful for things like, say, nWoD, where almost all of the roll is set, and you just have to tell it how many d10s to roll, explode on tens, and count successes of 8+.

**Comment by ArtemGr** [comment]
Appreciate the syntax example!

I plan to embed a WASM interpreter eventually, allowing users to program their own functions in Sidekick (in JavaScript/TypeScript, C++, Rust or whatever can compile to a small enough WASM).

In `/r $sixSidedDice X = Xd6` what worries me is a measure of extra ambiguity allowed for the variable names. E.g. what `d` means in `/r $name d = 1d6` ?

Right now you can do
```
/r $foo = 123
/r $bar = $foo + $foo
```

I'll probably tighten the syntax to allow for `${foo}d6`.

How about
```
/r $sixSidedDice ($x) = ${x}d6
/r $sixSidedDice (5)
```
Will the [sigil](https://en.wikipedia.org/wiki/Sigil_(computer_programming))ed variables work for you?
I doubt that normal (non-sigil) variables would be a good match for the [dice notation](https://en.wikipedia.org/wiki/Dice_notation).

**Comment by aturtledoesbite** [comment]
Yes, that format is just fine; the example I gave was just to show how I was thinking of it. I'm not picky.
